examples/specs/window_sum_waterfall_chart.vl.json

```diff
@@ -0,0 +1,104 @@
+{
+  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
+  "data": {
+    "values": [
+      {"label": "Start", "amount": 4000},
+      {"label": "Jan", "amount": 1707},
+      {"label": "Feb", "amount": -1425},
+      {"label": "Mar", "amount": -1030},
+      {"label": "Apr", "amount": 1812},
+      {"label": "May", "amount": -1067},
+      {"label": "Jun", "amount": -1481},
+      {"label": "Jul", "amount": 1228},
+      {"label": "Aug", "amount": 1176},
+      {"label": "Sep", "amount": 1146},
+      {"label": "Oct", "amount": 1205},
+      {"label": "Nov", "amount": -1388},
+      {"label": "Dec", "amount": 1492},
+      {"label": "End", "amount": 0}
+    ]
+  },
+  "width": 700,
+  "height": 450,
+  "transform": [
+    {"window": [{"op": "row_number", "as": "row"}]},
+    {
+      "window": [{"op": "sum", "field": "amount", "as": "sum"}],
+      "sort": [{"field": "row", "order": "ascending"}]
+    },
+    {
+      "calculate": "datum.label == 'End' ? 0 : datum.sum - datum.amount",
+      "as": "previous_amount"
+    },
+    {
+      "calculate": "datum.label == 'End' ? datum.sum : datum.amount",
+      "as": "amount"
+    },
+    {
+      "calculate": "datum.label == 'End' || datum.label == 'Start' ? 'Start/End' : datum.amount >= 0 ? 'increasing' : 'decreasing'",
+      "as": "Color"
+    },
+    {
+      "calculate": "datum.Color == 'decreasing' ? datum.sum : ''",
+      "as": "sum_dec"
+    },
+    {
+      "calculate": "datum.Color == 'increasing' ? datum.sum : ''",
+      "as": "sum_inc"
+    },
+    {
+      "calculate": "datum.Color == 'Start/End' ? datum.sum : ''",
+      "as": "sum_start_end"
+    }
+  ],
+  "encoding": {
+    "y": {
+      "field": "previous_amount",
+      "type": "quantitative",
+      "axis": {"title": "Amount"}
+    },
+    "y2": {"field": "sum", "type": "quantitative"},
+    "x": {
+      "field": "label",
+      "type": "ordinal",
+      "sort": {"field": "row", "op": "min"},
+      "axis": {"title": "Months"}
+    }
+  },
+  "layer": [
+    {
+      "mark": "bar",
+      "encoding": {
+        "color": {
+          "field": "Color",
```

Comment:
Instead of deriving a color field, I think it might be better to use conditions to set the color directly.